compatibility/build.go

```diff
@@ -0,0 +1,77 @@
+/*
+   Copyright 2020 The Compose Specification Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compatibility
+
+import "github.com/compose-spec/compose-go/types"
+
+func (c *WhiteList) CheckBuild(service *types.ServiceConfig) bool {
```

Comment:
Just catching my eye; should we `s/WhiteList/AllowList/`?

Reply:
Not being a native english speacker I can't really tell, but I know this approach to opt-in for a set of features as "WhiteList"

```golang
mychecker := WhiteList{
			Supported: []string{
				"services.network_mode",
				"services.privileged",
				"services.networks",
			}
```

Reply:
Yes, it's been commonly referred to as a "whitelist"; recent developments have raised questions wether "blacklist"/"whitelist" is a good term for these, or if "blocklist" / "denylist" / "allowlist" are more suitable.